contributing: minor release procedure updates (main)

neteler opened the pull request:
- markdown cosmetics and some outdated version numbers updated.
- fix path to helper script

Note: related PRs for the release branches have been created separately (do not backport this PR).

(replaces #2707)

wenzeslaus reviewed (approved):
Consider using sections instead of bold text if you think you need to navigate more easily in the document.

neteler commented:
Ok, I have now added subsections. 
I have also added a (tiny) "Publish release" section which was missing.

(see 3ab28bd)

I'll then update the related PRs accordingly once agreed here.

neteler commented:
If there are no objections, I'll merge this PR.
